### PR TITLE
TR-2938 - (UI) When publishing a template, immediately duplicating it before saving prevents duplication

### PR DESCRIPTION
**Pull request by nicklemmon**
[TR-2938](https://sparkpost.atlassian.net/browse/TR-2938)

### What Changed

- Initially added failing Cypress test that reproduced the bug - it passes now
- Cleaned up the templates create page Cypress tests to use more accurate queries and to appropriately wait for network requests to reduce the risk of flakiness
- Replaces use of `usePageFilters` for the template `id`, `version`, and `navKey` values with the [useParams](https://reactrouter.com/native/api/Hooks/useparams) hook from `react-router-dom`. It looks like this bug was introduced when we removed the custom `useRouter` hook from the app.
- Updates unit tests with appropriate mocks accordingly
- Moved domains tests within the `/configuration` directory
- Moved templates tests within the `/content` directory

### How To Test

- Follow the bug reproduction steps in the Jira ticket and verify the bug is resolved
- Test various redirects in the templates feature ensuring no breaking changes were introduced

### To Do

- [ ] Incorporate feedback

### CODE REVIEW

File: src/pages/templates/hooks/tests/useEditorNavigation.test.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
 import React from 'react';
 import { act } from 'react-dom/test-utils';
 import { mount } from 'enzyme';
+import { useParams } from 'react-router';
```
Comment (shrirupa) [inline src/pages/templates/hooks/tests/useEditorNavigation.test.js:4]: ahh, I always tend to import these hooks from reacct-router-dom. https://reactrouter.com/native/api/Hooks/useparams

**Comment by nicklemmon** [comment]
@shrirupa Updated that import - it worked like a charm. I'm not sure why but I had a comment indicating importing from `react-router-dom` wouldn't work - must have been another issue I eventually resolved.

File: src/pages/templates/hooks/tests/useEditorNavigation.test.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { act } from 'react-dom/test-utils';
 import { mount } from 'enzyme';
+import { useParams } from 'react-router-dom';
```
Comment (shrirupa) [inline src/pages/templates/hooks/tests/useEditorNavigation.test.js:4]: similar import fix here. sorry, should have found this one earlier review.

File: src/pages/templates/context/tests/EditorContext.test.js [hunk 3]
```diff
@@ -1,20 +1,28 @@
 import React from 'react';
 import { mount, shallow } from 'enzyme';
+import { useParams } from 'react-router';
```
Comment (shrirupa) [inline src/pages/templates/context/tests/EditorContext.test.js:3]: similar import fix here. sorry, should have found this one earlier review.

**Review by shrirupa — APPROVED** [review]
just that last import fix, everything else looks good to me!

**Pull request merged**